hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -223,6 +223,10 @@ def __call__(self, test):
                 'settings objects can be called as a decorator with @given, '
                 'but test=%r' % (test,)
             )
+        from hypothesis.stateful import GenericStateMachine
```

Comment:
Importing is relatively expensive, so I'd prefer to use a nested condition like the following:

```python
if isinstance(test, type):
    from hypothesis.stateful import GenericStateMachine
    if issubclass(test, GenericStateMachine):
        attr_name = '_hypothesis_internal_settings_already_applied'
        if getattr(test, attr_name, False):
            raise InvalidArgument(
               'Applying the @settings decorator twice would overwrite '
               'the first version; merge their arguments instead.'
            )
        test.TestCase.settings = self
        setattr(test, attr_name, True)
        return test
    else:
        raise InvalidArgument(
            '@settings(...) can only be used as a decorator on '
            'functions or subclasses of GenericStateMachine.'
        )
```

Note that the new explicit error is backwards-compatible, as this has never worked - we just want to add an explicit error because it now *does* work for *some* classes.  It will also need a covering test.

Note also that I've replicated the check for if settings have already been applied, but becuase this has never been valid before we can make it an error immediately.  This will also need a test!